src/lib/salad/mhash.h

```diff
@@ -75,6 +79,26 @@
 
 #define mh_unlikely(x)  __builtin_expect((x),0)
 
+#define mh_mmi(p) (UINT64_C(0xFFFFFFFFFFFFFFFF) / (p) + 1)
+
+#if MH_FAST_MOD
+#define mh_mul_u64_u32(u64, u32) \
+	({ (uint32_t)(((__uint128_t)(u64) * (u32)) >> 64); })
```

Comment:
__uint128_t is extension and may be not supported by compiler. I guess we need extra checks here.

Reply:
This compiles to **GCC** version as old as [4.1.2](https://godbolt.org/z/oaoo9jqTs), while the oldest compiler we support is **GCC** 4.8 on CentOS, so I believe this is nitpicking.

Reply:
Fixed: added a conditional compilation check for this compiler feature.